tests/nested/manual/grade-signed-cloud-init-testkeys/task.yaml

```diff
@@ -0,0 +1,166 @@
+  # prepare-device hook to use the fakedevicesvc
+  mkdir extra-snaps
+
+  # Get the snakeoil key and cert for signing gadget assets (shim)
+  KEY_NAME=$(nested_get_snakeoil_key)
+  SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+  SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+
+  snap download --basename=pc --channel="20/edge" pc
+  unsquashfs -d pc-gadget pc.snap
+
+  # delay all refreshes for a week from now, as otherwise refreshes for our 
+  # snaps (which are asserted by the testrootorg authority-id) may happen, which
+  # will break things because the signing keys won't match, etc. and 
+  # specifically snap-bootstrap in the kernel snap from the store won't trust
+  # the seed keys to unlock the encrypted data partition in the initramfs
+  sed defaults.yaml -e "s/HOLD-TIME/$(date --date='next week' +%Y-%m-%dT%H:%M:%S%:z)/" >> \
+    pc-gadget/meta/gadget.yaml
+
+  # TODO: enable this bit when things are ready to use a testkeys signed model
+  # assertion
+  # copy the prepare-device hook to use our fakedevicesvc
+  # mkdir -p pc-gadget/meta/hooks/
+  # cp prepare-device pc-gadget/meta/hooks/
+
+  nested_secboot_sign_gadget pc-gadget "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  snap pack pc-gadget/ extra-snaps/
+
+  rm -f "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+
+  # TODO: enable when ready
+  # start fake device svc
+  # systemd_create_and_start_unit fakedevicesvc "$(command -v fakedevicesvc) localhost:11029"
+
+  # first boot - legit NoCloud usage
+  nested_build_seed_cdrom "$TESTSLIB/cloud-init-seeds/normal-user" seed.iso cidata user-data meta-data
+
+  # second boot - attacker drive
+  nested_build_seed_cdrom "$TESTSLIB/cloud-init-seeds/attacker-user" seed2.iso cidata user-data meta-data
+
+  # first boot will use seed1 to create the normal-user in addition to the 
+  # system-user assertion
+  NESTED_PARAM_CD="-cdrom $(pwd)/seed.iso"
+
+  nested_create_core_vm
+  nested_start_core_vm
+
+restore: |
+  if [ "$TRUST_TEST_KEYS" = "false" ]; then
+      echo "This test needs test keys to be trusted"
+      exit
+  fi
+
+  #shellcheck source=tests/lib/store.sh
+  . "$TESTSLIB"/store.sh
+  teardown_fake_store "$NESTED_FAKESTORE_BLOB_DIR"
+
+execute: |
+  if [ "$TRUST_TEST_KEYS" = "false" ]; then
+      echo "This test needs test keys to be trusted"
+      exit
+  fi
+
+  #shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+
+  echo "The initial cloud-init user was created"
+  nested_exec "cat /var/lib/extrausers/passwd" | MATCH normal-user
+
+  echo "And we can run things as the normal user"
+  nested_exec_as normal-user ubuntu "sudo true"
+
+  # TODO: check that we got a serial assertion via the fakedevicesvc
+  # for now we just don't get a serial assertion which is fine for the purposes
+  # of this test
+
+  # TODO: is there a better thing we can wait for here instead? maybe the log
+  # message from snapd directly via journalctl ?
+  echo "Waiting for snapd to react to cloud-init"
+  sleep 60
```

Comment:
:(

Reply:
haha yeah it's on my queue to make this more robust, but I want to land this test first